main.js

```diff
@@ -0,0 +1,327 @@
+
+  $.ajax({
+    method: "GET",
+    url: searchURL,
+    dataType: "json",
+    success: function (data) {
+      successCB(data)
+    },
+    error: function (jqXHR, textStatus, errorThrown) {
+      console.log(textStatus);
+      alert("Search did not return valid result. Please try again.")
+    },
+  })
+}
+
+//Fills the appState.weatherInfo object with the relevant current conditions data
+var setWeatherInfo = function(OpenWeatherdata) {
+  appState.weatherInfo = {};
+  appState.weatherInfo.city = OpenWeatherdata.name;
+  appState.weatherInfo.country = OpenWeatherdata.sys.country;
+  //Data from open weather is in Kelvin: Kelvin to Fahrenheit conversion: (temp − 273.15) × 9/5 + 32
+  appState.weatherInfo.temperature = Math.round((OpenWeatherdata.main.temp-273.15)*9/5+32);
+  appState.weatherInfo.weatherConditions = OpenWeatherdata.weather[0].description;    //weather[0].main;
+  appState.weatherInfo.weatherConditionsIcon = `http://openweathermap.org/img/wn/${OpenWeatherdata.weather[0].icon}@2x.png`;
+}
+
+//Sets the location that weather is being displayed for.
+var setLocation = function(OpenWeatherData) {
+  appState.location = {};
+  appState.location = {
+    type: "Coordinates",
+    latitude: OpenWeatherData.coord.lat,
+    longitude: OpenWeatherData.coord.lon,
+    city: OpenWeatherData.name
+  }
+}
+
+//Displays Current Conditions portion of page
+var renderCurrentConditions = function(state) {
+  var source = $("#current-conditions-template").html();
+  var template = Handlebars.compile(source);
+  var currentConditionsHTML = template(state.weatherInfo);
+  $(".current-weather").html(currentConditionsHTML);
+  $("#city").val('')
+}
+
+//Creates and Displays a map centered on the location weather is being displayed for
+var createMap = function (state) {
+  var source = $("#map-template").html();
+  var template = Handlebars.compile(source);
+  var mapHTML = template(state.location);
+  $(".map").html(mapHTML);
+}
+
+
+//5 day forecast functions
+//Gets needed weather data from Open Weather API for 5 day forecast
+var fetchFiveDayForecast = function(location, successCB) {
+  var searchURL = '';
+  searchURL = location.type === "City Name" ? `https://api.openweathermap.org/data/2.5/forecast?q=${location.city}&appid=59b871a25f174e2019ec1f4fbbe6807c` : searchURL;
+  searchURL = location.type === "Coordinates" ? `https://api.openweathermap.org/data/2.5/forecast?lat=${location.latitude}&lon=${location.longitude}&appid=59b871a25f174e2019ec1f4fbbe6807c` : searchURL;
+
+  $.ajax({
+    method: "GET",
+    url: searchURL,
+    dataType: "json",
+    success: function (data) {
+      successCB(data);
+    },
+    error: function (jqXHR, textStatus, errorThrown) {
+      console.log(textStatus);
+    },
+  })
+}
+
+//Sets the appState.forecastInfo array with the relevant 5 day forecast data
+var setForecastInfo = function(OpenWeatherdata) {
+  appState.forecastInfo = [{},{},{},{},{}];
```

Comment:
It isn't common to see something like this - an array with a bunch of hardcoded empty objects